algorithms/src/snark/varuna/ahp/prover/round_functions/first.rs

```diff
@@ -57,7 +57,6 @@ impl<F: PrimeField, MM: SNARKMode> AHPForR1CS<F, MM> {
     }
 
     /// Output the first round message and the next state.
-    #[allow(clippy::type_complexity)]
```

Comment:
I imagine this and a lot of other lints are artifacts from a time when we didn't have the types encapsulated as they are now. There's no "redundant clippy lint" lint, so these can pile up. Thanks for removing them!